common/src/main/java/com/wynntils/handlers/inventory/InventoryHandler.java

```diff
@@ -147,41 +170,62 @@ public void onWorldStateChange(WorldStateEvent event) {
+        if (menu.slots.size() > 36 && menu instanceof ChestMenu) {
+            for (int slotNum = menu.slots.size() - 37; slotNum >= 0; slotNum--) {
+                if (ItemUtils.isNonSlotPlaceholder(menu.getSlot(slotNum).getItem())) {
+                    // Click the placeholder item, which should be no-op, but which will prompt an update packet
+                    McUtils.sendPacket(new ServerboundContainerClickPacket(
+                            menu.containerId,
+                            menu.getStateId(),
+                            slotNum,
+                            0,
+                            ClickType.PICKUP,
+                            ItemStack.EMPTY,
+                            Int2ObjectMaps.emptyMap()));
+                    return;
+                }
             }
         }
 
-        // Find a nonempty inventory slot
-        for (int slotNum = 9; slotNum <= menu.slots.size(); slotNum++) {
-            ItemStack stack = menu.getSlot(slotNum).getItem();
-            if (!stack.isEmpty()) {
-                // Swap the item into hotbar slot 0, which should be no-op, but which will prompt an update packet
-                McUtils.sendPacket(new ServerboundContainerClickPacket(
-                        menu.containerId,
-                        menu.getStateId(),
-                        slotNum,
-                        0,
-                        ClickType.SWAP,
-                        ItemStack.EMPTY,
-                        Int2ObjectMaps.emptyMap()));
-                return;
+        // Method 2: swap the content book into an empty slot
+        ItemStack contentBookStack = McUtils.player().getInventory().getItem(8);
+        if (!contentBookStack.isEmpty()) {
+            for (int slotNum = 0; slotNum <= menu.slots.size(); slotNum++) {
+                Slot slot = menu.slots.get(slotNum);
+                if (slot.getItem().isEmpty() && slot.mayPlace(contentBookStack)) {
+                    // Swap the book into the slot, which should be no-op, but which will prompt an update packet
+                    McUtils.sendPacket(new ServerboundContainerClickPacket(
+                            menu.containerId,
+                            menu.getStateId(),
+                            slotNum,
+                            8,
+                            ClickType.SWAP,
+                            ItemStack.EMPTY,
+                            Int2ObjectMaps.emptyMap()));
+                    return;
+                }
             }
         }
-        // At this point, our view of the inventory must be completely empty; there's not much we can do
+
+        // Method 3: swap the ingredient pouch into the content book
+        // This is the method of last resort because in certain inventories with filtered interactions (e.g. housing
+        // inventories or blacksmiths), swapping the pouch will prompt an annoying error message in the chat
```

Comment:
are u sure this is a good way to do this? Afaik, opening the ingredient pouch has to be blocked in some inventories, as it can override reward UIs for example, essentially voiding the loot inside.